### DISCUSSION TITLE
RPC-Server - Can't set server to TestNet or MainNet

### DISCUSSION THREAD

**Discussion by Thomas-Bibby** [reach-sh/reach-lang#1229] [Q&A] [answered]
I'm running a rpc-server locally using ./reach rpc-server and it is only in dev-net. How do I run an rpc server in TestNet or MainNet?

I have also tried running this and using `var walletFallback = await rpc.Call("/stdlib/walletFallback", Client.AsJson("{providerEnv: 'TestNet'}"));` on my c# backend to change it to testnet but was given an error.

`System.Text.Json.JsonException: ''p' is an invalid start of a property name. Expected a '"'. Path: $ | LineNumber: 0 | BytePositionInLine: 1.'
`

I already can do other commands like parseCurrency so its not an issue connecting to the server.

**Comment by DanBurton** [comment, marked as answer]
Try the `/stdlib/setProviderByName` call instead, and just give it the string `"TestNet"`.

**Reply by Thomas-Bibby** [reply]
Same error as above

**Reply by jeapostrophe** [reply]
You did `await rpc.Call("/stdlib/setProviderByName", "TestNet")` and got an error about the character `p`?

**Reply by Thomas-Bibby** [reply]
Well yes but no, I got the error about character T not p but otherwise the same 
![image](https://user-images.githubusercontent.com/62004466/177003383-b0e201c0-ebd1-4ebd-b2db-341a9e12092b.png)

**Reply by jeapostrophe** [reply]
I have now actually looked at a C# example. Did you do `await rpc.Call("/stdlib/setProviderByName", Client.AsJson("TestNet"))`? FWIW, I can't find an example of us calling the C# encoder with a string or using this function, so I'm suspecting it is something wrong with the client interface rather than the RPC server itself

**Reply by Thomas-Bibby** [reply]
Yeah I called `await rpc.Call("/stdlib/setProviderByName", Client.AsJson("TestNet"))` and it seems to be an issue with passing a string as if I pass a number like 100 it still errors but at a later point. Is there any other way to use the rpc-server on test or mainnet?

**Reply by jeapostrophe** [reply]
I think the correct call MIGHT be `await rpc.Call("/stdlib/setProviderByName", Client.AsJson("\"TestNet\""))`

**Reply by Thomas-Bibby** [reply]
That works thank you.

**Reply by jeapostrophe** [reply]
Phew!
